### PR TITLE
chore(EMS-3387): iterations - move application creation logic into helpers

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:
This PR moves some application creation logic away from the KeystoneJS context hooks (`resolveInput`, `afterOperation`) and into our own, existing application creation helpers.

The KeystoneJS context hooks can be very helpful, but in this instance, it could be confusing for new developers to understand and maintain.

Also, by using our own helpers/logic instead of "hooking" into the KeystoneJS context hooks, this means that:
- All application creation logic is in helpers.
- Application creation is simpler.
- There are less database operations involved.
- Vastly improved logs/debugging.
- The `createFullApplication` test helper is vastly simplified.

This PR also introduces a variety of documentation and test description alignments.

## Resolution :heavy_check_mark:
- Create new helper functions:
  - `createABusiness`
  - `createADeclaration`
  - `createAPolicyContact`
  - `createAReferenceNumber`
- Update `createApplicationRelationships` helper function to:
  - Create a broker.
  - Create a business.
  - Create a declaration.
  - Create a policy contact.
  - Return additional relationship IDs.
- Update `createInitialApplication` helper function to create default data (deal type, submission count, deadline, etc).
- Update `createAnApplication` helper function to obtain and pass additional relationship IDs.
- Update `updateApplicationColumns` helper function to consume and pass additional relationship IDs.
- Remove application KeystoneJS hooks from `schema.ts`. This is now handled in the above helper functions.

## Miscellaneous :heavy_plus_sign:
- Various documentation improvements and alignments.
- Various test description alignments.
- Replace instances of "blank" with "empty".
- Various logging improvements.
- Update some instances of `createAnApplication` that were using a helper to be called `createAnApplicationHelper`, to avoid confusion with the `createAnApplication` GQL mutation.

### CODE REVIEW

File: src/api/custom-resolvers/queries/get-ordnance-survey-address/index.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ const getOrdnanceSurveyAddress = async (root: any, variables: OrdnanceSurveyVari
 
     const response = await ordnanceSurvey.get(postcode);
 
-    // if no data in response or status is not 200 then return blank object
+    // if no data in response or status is not 200 then return empty object
```
Comment (Zainzzkk) [inline src/api/custom-resolvers/queries/get-ordnance-survey-address/index.ts:36]: return an?

File: src/api/custom-resolvers/queries/get-companies-house-information/index.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ const getCompaniesHouseInformation = async (root: any, variables: GetCompaniesHo
 
     const response = await companiesHouse.get(sanitisedNumber);
 
-    // if no data in response or status is not 200 then return blank object
+    // if no data in response or status is not 200 then return empty object
```
Comment (Zainzzkk) [inline src/api/custom-resolvers/queries/get-companies-house-information/index.ts:25]: return an?

File: src/api/test-helpers/loss-payee-financial-international-vector.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { ApplicationLossPayeeFinancialInternationalVector, TestHelperLossPayeeFi
 /**
  * createLossPayeeFinancialDetailsInternationalVector
  * create a nominated loss payee financial international helper
- * Creates a blank ApplicationLossPayeeFinancialInternationalVector.
+ * Creates a empty ApplicationLossPayeeFinancialInternationalVector.
```
Comment (Zainzzkk) [inline src/api/test-helpers/loss-payee-financial-international-vector.ts:6]: an empty

File: src/api/test-helpers/loss-payee-financial-uk-vector.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@ import { ApplicationLossPayeeFinancialUkVector, TestHelperLossPayeeFinancialUkVe
 /**
  * createLossPayeeFinancialDetailsUkVector
  * create a nominated loss payee financial uk helper
- * Creates a blank ApplicationLossPayeeFinancialUkVector.
+ * Creates a empty ApplicationLossPayeeFinancialUkVector.
```
Comment (Zainzzkk) [inline src/api/test-helpers/loss-payee-financial-uk-vector.ts:6]: same as above

File: src/api/test-helpers/nominated-loss-payee.ts [hunk 5]
```diff
@@ -4,8 +4,8 @@ import { ApplicationNominatedLossPayee, ApplicationLossPayeeFinancialUk, Applica
 
 /**
  * create a nominated loss payee financial uk helper
- * Creates a blank ApplicationLossPayeeFinancialUk.
- * @param {Context} KeystoneJS context API
+ * Creates a empty ApplicationLossPayeeFinancialUk.
```
Comment (Zainzzkk) [inline src/api/test-helpers/nominated-loss-payee.ts:7]: same as above

**Comment by ttbarnes** [comment]
Thanks @Zainzzkk good spot, i've fixed those documentaitons - will push shortly with some unit test fixes 👍

File: src/ui/server/helpers/https/index.ts [hunk 6]
```diff
@@ -7,7 +7,7 @@ dotenv.config();
 const { PORT } = process.env;
 
 /**
- * Creates an HTTPS server using the `https` module.
+ * Create ann HTTPS server using the `https` module.
```
Comment (Zainzzkk) [inline src/ui/server/helpers/https/index.ts:10]: typo

**Review by Zainzzkk — APPROVED** [review]

**Pull request merged**